contracts/src/v0.8/keystone/KeystoneForwarder.sol

```diff
@@ -15,19 +15,55 @@ contract KeystoneForwarder is IForwarder, ConfirmedOwner, TypeAndVersionInterfac
   /// @param data the data that was received
   error InvalidData(bytes data);
 
+  error RepeatedSigner();
+
+  error WrongNumberOfSignatures();
+
+  error InvalidSigner();
+
   uint256 private constant SELECTOR_LENGTH = 4;
   uint256 private constant REPORT_LENGTH = 64;
 
   struct HotVars {
     bool reentrancyGuard; // guard against reentrancy
+    uint8 n;
+    uint8 f;
   }
 
   HotVars internal s_hotVars; // Mixture of config and state, commonly accessed
 
+  /// @notice Contains the signing address of each oracle
+  address[] internal s_signers;
```

Comment:
Signers are per DON ID, so this should be a mapping.

Reply:
https://docs.google.com/document/d/1VlosqVXzJhWP77ss4UOH0b4rFX2J4sgfdzt0-prHHw0/edit?disco=AAABMI3rMsc